packages/epics/src/agreements/plugins/payment-schedule/payment-schedule.tsx

```diff
@@ -0,0 +1,221 @@
+'use client';
+
+import {
+  useForm,
+  Controller,
+  useFieldArray,
+  FormProvider,
+  useFormContext,
+} from 'react-hook-form';
+import {
+  DropdownMenu,
+  DropdownMenuTrigger,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  Button,
+  Label,
+  Input,
+  DatePicker,
+} from '@hypha-platform/ui';
+import { cn } from '@hypha-platform/lib/utils';
+import { ChevronDownIcon, PlusIcon, Cross2Icon } from '@radix-ui/react-icons';
+import { PercentIcon } from 'lucide-react';
+import {
+  validateMilestones,
+  validateFutureDate,
+  DateRange,
+  Milestone,
+} from './validation';
+
+const options = ['Immediately', 'Future Payment', 'Milestones'] as const;
+type Option = (typeof options)[number];
+
+export type FormValues = {
+  option: Option;
+  futureDate?: Date;
+  milestones: Milestone[];
+};
+
+function useProvidedOrLocalForm() {
+  const context = useFormContext<FormValues>();
+  const isInsideProvider = !!context?.control;
+  const methods = useForm<FormValues>({
+    defaultValues: {
+      option: 'Immediately',
+      futureDate: undefined,
+      milestones: [],
+    },
+    mode: 'onChange',
+  });
+
+  return { methods, isInsideProvider };
```

Comment:
I think we can define that these are composable form components that have to be used within a form context. Then we get rid of this conditional.